analyzers/src/SonarAnalyzer.CSharp/Rules/CloudNative/AzureFunctionsCatchExceptions.cs

```diff
@@ -0,0 +1,105 @@
+{
+    public static class Fixme   // FIXME: Remove temporary workaround
+    {
+        public static bool IsAzureFunction(this SyntaxNodeAnalysisContext context) =>
+            context.ContainingSymbol.GetAttributes().Any(x => x.AttributeClass.Name.Contains("FunctionName"));
+    }
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class AzureFunctionsCatchExceptions : SonarDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S6421";
+        private const string MessageFormat = "Wrap Azure Function body in try/catch block.";
+
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        protected override void Initialize(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+                {
+                    if (c.IsAzureFunction())
+                    {
+                        var method = (MethodDeclarationSyntax)c.Node;
+                        var walker = new Walker();
+                        if (walker.SafeVisit(c.Node) && walker.HasInvocationOutsideTryCatch)
+                        {
+                            c.ReportIssue(Diagnostic.Create(Rule, method.Identifier.GetLocation()));
+                        }
+                    }
+                },
+                SyntaxKind.MethodDeclaration);
+
+        private sealed class Walker : SafeCSharpSyntaxWalker
+        {
+            public bool HasInvocationOutsideTryCatch { get; private set; }
+
+            public override void Visit(SyntaxNode node)
+            {
+                if (!HasInvocationOutsideTryCatch)  // Stop walking when we know the answer
+                {
+                    switch (node.Kind())
+                    {
+                        case SyntaxKind.AnonymousMethodExpression:
+                        case SyntaxKind.SimpleLambdaExpression:
+                        case SyntaxKind.ParenthesizedLambdaExpression:
+                        case SyntaxKindEx.LocalFunctionStatement:
+                            return;
+                    }
+                    base.Visit(node);
+                }
+            }
+
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node) =>
+                HasInvocationOutsideTryCatch = true;
+
+            public override void VisitTryStatement(TryStatementSyntax node)
+            {
+                if (!node.Catches.Any(CatchesAllExceptions))
+                {
+                    base.VisitTryStatement(node);
+                }
+            }
+
+            public override void VisitCatchClause(CatchClauseSyntax node)
+            {
+                // Do not visit content of "catch". It doesn't make sense to wrap logging in catch in another try/catch.
+            }
```

Comment:
Consider adding SyntaxKind.CatchClause to the bailout list in the `Visit` method instead.